src/App.tsx

```diff
@@ -1,28 +1,66 @@
-import { useSelector } from 'react-redux';
-
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { getTodos } from './api';
 import './App.scss';
-import { Start } from './components/Start';
-import { Finish } from './components/Finish';
+import { Loader } from './components/Loader/Loader';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
+import { TodoModal } from './components/TodoModal';
 
-import { selectors } from './store';
+import { useAppSelector } from './store';
+import { actions as loadingActions } from './store/loading';
+import { actions as todosActions } from './store/todos';
 
 export const App = () => {
-  // `useSelector` connects our component to the Redux store
-  // and rerenders it after every dispatched action
-  const loading = useSelector(selectors.isLoading);
+  const loading = useAppSelector(state => state.loading);
+  const dispatch = useDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    dispatch(loadingActions.startLoading());
 
-  // we do not call a selector with (), just pass a link to it
-  const message = useSelector(selectors.getMessage) || 'Ready!';
+    getTodos()
+      .then((todosFromServer) => dispatch(
+        todosActions.setTodos(todosFromServer),
+      ))
+      .finally(() => dispatch(
+        loadingActions.finishLoading(),
+      ));
+  }, []);
 
   return (
     <div className="App">
       <h1>Redux list of todos</h1>
-      <h2>{loading ? 'Loading...' : message}</h2>
+      <>
+        <div className="section">
+          <div className="container">
+            <div className="box">
+              <h1 className="title">Todos:</h1>
+
+              <div className="block">
+                <TodoFilter />
+              </div>
+
+              <div className="block">
+                { loading ? (
+                  <Loader />
+                )
+                  : (
+                    <TodoList />
+                  )}
```

Comment:
```suggestion
              <div className="block">
                { loading 
                  ? <Loader />
                  : <TodoList />
                }
```

GJ 🔥 
Everything is great! There just few moments that hard to read.